ledger/acctupdates.go

```diff
@@ -888,6 +888,13 @@ type accountUpdatesLedgerEvaluator struct {
 	// prevHeader is the previous header to the current one. The usage of this is only in the context of initializeCaches where we iteratively
 	// building the ledgercore.StateDelta, which requires a peek on the "previous" header information.
 	prevHeader bookkeeping.BlockHeader
+	// blockTimeStamps contains block timestamps needed for txn evaluation while catching account db round up to block db round.
+	// in this mode au uses accountUpdatesLedgerEvaluator as a ledger, that is enough because balances are available in the underlying accountdb.
+	// unfortunately this does not work anymore after allowing smart contact accessing block header timestamps up to MaxTxnLife.
+	// block db and TxTail has required info (TxTail tracker provides timestamps in a normal evaluation mode)
+	// but calling TxTail from accountUpdatesLedgerEvaluator introduces trackers' dependency and locking problems.
+	// instead, caching timestamps in the same manner as TxTail prevents dependency and ledgerForTracker interface pollution.
+	blockTimeStamps map[basics.Round]int64
```

Comment:
I don't understand everything here perfectly, but it seems like info in txtail, like leases, is just as important for transaction processing.  So why is txtail "good enough" with respect to leases, but won't have the info we need for timestamps?

Reply:
these are used in different code paths: leases, txn ids and timestamps are by evaluator that has entire Ledger with all trackers. This one is only used for "catching" up balances to the same round as block DB in order to accept new transactions. It is done within account updates tracker, and leases/ids do not matter since they come from block DB that are already persisted in blockchain. But now evaluation requires block timestamps.

Reply:
The above logical distinction might suggest that we should maintain these on separate trackers. But we can refactor that on a separate PR.